### ISSUE TITLE
Get better at setting focus on editor at the right times

### ISSUE THREAD

**Issue by dm-p** [deneb-viz/deneb#344]
In an attempt to be more user-friendly with the editor component, we take a shotgun to set focus on it, particularly when buttons are clicked. This is not very accessible and creates issues with tabbing around the interface. As we're looking to improve the ability to get out of the editor to be able to navigate to other elements, we should aim for the following story:

- [x] When the editor is opened, the focus should be set automatically on the active one.
- [x] For anything else, we don't set focus back to the editor, except for (and these are subject to change pending review):
    - [x] Closing the _Create new specification_ dialog
    - [x] Closing the _Edit specification field mapping_ dialog
    - [x] Closing the _Generate JSON template_ dialog
- [x] Set focus to the editor only when the Spec/Config pane and Apply and Fix buttons are invoked
